synapse/http/site.py

```diff
@@ -333,14 +333,28 @@ def _should_log_request(self) -> bool:
 
 
 class XForwardedForRequest(SynapseRequest):
-    def __init__(self, *args, **kw):
-        SynapseRequest.__init__(self, *args, **kw)
-
     """
     Add a layer on top of another request that only uses the value of an
     X-Forwarded-For header as the result of C{getClientIP}.
+
+    XXX: I think the right way to do this is with request.setHost().
     """
 
+    def __init__(self, *args, **kw):
+        SynapseRequest.__init__(self, *args, **kw)
+
+        forwarded_header = self.getHeader(b"x-forwarded-proto")
```

Comment:
what do people think about this? we don't currently document that this is a thing that ought to be set (and indeed, matrix.org's haproxy doesn't). Should we? or are there better ways to solve this problem?

Reply:
I think this is pretty standard -- I don't know of another way to do this (and I don't see a security issue if this was sent directly via a client, it might break things but not in a "bad" way?)